test/Libraries/Microsoft.Extensions.AI.Abstractions.Tests/AssertExtensions.cs

```diff
@@ -57,7 +58,9 @@ private static void AreJsonEquivalentValues(object? expected, object? actual, Js
         options ??= JsonSerializerOptions.Default;
         JsonElement expectedElement = NormalizeToElement(expected, options);
         JsonElement actualElement = NormalizeToElement(actual, options);
-        if (!JsonElement.DeepEquals(expectedElement, actualElement))
+        if (!JsonNode.DeepEquals(
```

Comment:
Note that the pre-.NET 9 version of that implementation is very rough and can result in false negative comparisons.

Reply:
I think we could still up the STJ version to v9 for net8.0 tests without losing coverage.

Reply:
I actually see this as a benefit. All our testing was being done against STJ v9, so if there were any STJ v8 bugs that the implementation might hit, our testing on .NET 8 and .NET Framework wouldn't reveal it.

Reply:
> Note that the pre-.NET 9 version of that implementation is very rough and can result in false negative comparisons.

We run all tests on .NET Framework, .NET 8, and .NET 9. We'll get coverage with both STJ v8 on the former two and STJ v9 on the latter.